pkgs/applications/networking/cato/default.nix

```diff
@@ -0,0 +1,63 @@
+{
+  stdenv,
+  fetchurl,
+  dpkg,
+  lib,
+  libz,
+  autoPatchelfHook,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "cato";
+  version = "5.2.1.1";
+
+  src = fetchurl {
+    url = "https://clients.catonetworks.com/linux/${version}/cato-client-install.deb";
+    sha256 = "sha256-0hUchaxaiKJth2ByQMFfjsCLi/4kl+SrNSQ33Y6r3WA=";
+  };
+
+  dontConfigure = true;
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+  ];
+
+  buildInputs = [
+    libz
+    stdenv.cc.cc
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+    dpkg -x $src source
+    cd source
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    mkdir $out
+    mv etc $out/etc
```

Comment:
Can you elaborate on why etc needs to be copied into this derivation?

Reply:
```
unpacked/
├── etc
│   └── ld.so.conf.d
│       └── cato-client.conf
├── lib
│   └── systemd
│       └── system
│           └── cato-client.service
└── usr
    ├── bin
    │   └── cato-sdp
    ├── lib
    │   └── cato
    │       ├── libwaapi.so -> libwaapi.so.4
    │       ├── libwaapi.so.4 -> libwaapi.so.4.3.2690.0
    │       ├── libwaapi.so.4.3.2690.0
    │       ├── libwacollector.so -> libwacollector.so.4
    │       ├── libwacollector.so.4 -> libwacollector.so.4.3.2690.0
    │       ├── libwacollector.so.4.3.2690.0
    │       ├── libwadeviceinfo.so -> libwadeviceinfo.so.4
    │       ├── libwadeviceinfo.so.4 -> libwadeviceinfo.so.4.3.2690.0
    │       ├── libwadeviceinfo.so.4.3.2690.0
    │       ├── libwaheap.so -> libwaheap.so.4
    │       ├── libwaheap.so.4
    │       ├── libwalocal.so -> libwalocal.so.4
    │       ├── libwalocal.so.4 -> libwalocal.so.4.3.2690.0
    │       ├── libwalocal.so.4.3.2690.0
    │       ├── libwaresource.so
    │       ├── libwautils.so -> libwautils.so.4
    │       ├── libwautils.so.4 -> libwautils.so.4.3.2690.0
    │       ├── libwautils.so.4.3.2690.0
    │       ├── license.cfg
    │       └── wadiagnose
    ├── local
    │   ├── lib
    │   └── share
    │       └── ca-certificates
    │           └── CatoNetworksTrustedRootCA.crt
    └── sbin
        └── cato-clientd

15 directories, 25 files
```

This is the contents of the deb. I think the client can download its own certificate on startup, so perhaps there's no need to copy usr/local, i'm not even sure if those paths (etc and usr/local) are available for it at runtime.
I'll do some more thinking, and look at the systemd config too. Thanks for your review

Reply:
Having said that, the contents of cato-client.conf is just the following line:
```
/usr/lib/cato
```
I think it can be safely binned :joy: